plugin/storage/es/dependencystore/storage.go

```diff
@@ -37,33 +37,43 @@ const (
 
 // DependencyStore handles all queries and insertions to ElasticSearch dependencies
 type DependencyStore struct {
-	ctx         context.Context
-	client      es.Client
-	logger      *zap.Logger
-	indexPrefix string
+	ctx                 context.Context
+	client              es.Client
+	logger              *zap.Logger
+	indexPrefix         string
+	useReadWriteAliases bool
 }
 
 // NewDependencyStore returns a DependencyStore
-func NewDependencyStore(client es.Client, logger *zap.Logger, indexPrefix string) *DependencyStore {
+func NewDependencyStore(client es.Client, logger *zap.Logger, indexPrefix string, useReadWriteAliases bool) *DependencyStore {
```

Comment:
We should introduce params struct

Reply:
Urgh, I almost had that done and was wondering why the storage modules were so different. But I thought to make it a small as possible. But got point, I shall add a struct carrying the config as with the spans.

Do you also want to split reader and writer?

Reply:
We can keep them together if there is no requirement to split them.